### ISSUE TITLE
Hi, it's been a while

### ISSUE THREAD

**Issue by ddavness** [ddavness/power-mailinabox#139]
As you've noticed, it's been a while since I last touched this project. Alas, life decided to do it's thing. I now have a job, yadda yadda, and am pursuing other projects.

Yet, I'm not intending to leave the project abandoned just yet.

## What now from here?

It will be an herculean effort, since we're like 8 versions behind. Alongside that, Debian users are still required to sit on Debian 11; I want to add support to it along the way. Definitely expect this to take a while to complete!

It is definitely possible that the catch-up to the upstream version might be done gradually. Stay tuned for updates! 👋

## And after that?

We'll see what the future holds. To keep workload low from my side, new features won't be actively worked on anymore by myself. Pull Requests will still remain welcome, naturally.

**Comment by ddavness** [comment]
Bit of a note here:
- Debian 11 runs on PHP 7.4
- Debian 12 runs on PHP 8.2

There is no Nextcloud version that supports both PHP versions (Up to v24 on Debian 11, v25 and beyond on Debian 12). This means there will likely be a stop-gap update along in the way (i.e. we won't be able to upgrade all the way to v68).

Still unsure how to do this while making sure that a certain release can be installed in both Debian 11 and Debian 12 :thinking:

**Comment by kiekerjan** [comment]
Perhaps you can use packages from https://sury.org/ to install the php version you need? This seems to be the php maintainer from Debian. (needs to be confirmed though)

**Comment by ddavness** [comment]
Yeah - to be fair I would really prefer if this didn't involve using third party repositories. But I'm not seeing another choice 😔

**Comment by x-FBi** [comment]
@ddavness any word on progression?  We're a few revisions behind the main release.

**Comment by ddavness** [comment]
Apologies, was catching up to life (or perhaps, life was catching up to me! :sweat_smile:)
Going to resume working on this. Noticed the mainstream is at v69 now (!)

After thinking on this for a bit, I've decided that for the PHP issue, the stop-gap version will install a third-party repository on distros that do not support NC 25 out of the box (i.e. Debian 11 and Ubuntu 20.04), and then remove it* once the user upgrades to Debian 12/Ubuntu 22.04

\* On Debian, the best way to do this is to create a `apt/sources.list.d/mailinabox.list` file. That way if the user intentionally adds the same repository anywhere else, it'll remain there; Not sure if we can compartimentalize like this on Ubuntu (where `add-apt-repository` is "king". I'll have to investigate.)

**Comment by ddavness** [comment]
After some investigation, I'm leaning towards the following:

The stopgap version will be based on **[v67](https://github.com/mail-in-a-box/mailinabox/blob/main/CHANGELOG.md#version-67-december-22-2023)**, which is the latest upstream version that installs NextCloud 25 (the last one to still support PHP 7.4 - even if deprecated)

On that stopgap version:
* Will be the last version supported by Ubuntu 20.04 - **it will still use PHP 7.4, no PPA's will be added**
* Support for Ubuntu 22.04 will be enabled
* Support for Debian 12 will **NOT** be enabled yet.

From there, we skip straight ahead to the latest upstream version (currently **v69**), which is running Nextcloud 26:
* Will be the last version to be supported by Debian 11 - **a third-party repository will be installed to bump PHP to version 8.0 in order to ease the transition**. With this said:
* * The earliest version of PMiaB supported by Debian 11 is **[v55.0](https://github.com/ddavness/power-mailinabox/releases/tag/v55.0)**, which ships with NextCloud 20. It does not support PHP 8.0
* * The first version of NC supporting PHP 8.0 is NC 21. I'll need to do some testing to confirm whether it's possible to do a migration from NC 20 all the way to NC 26 on PHP 8.0 alone. (i.e. go from **v55.0** to a theoretical v69.0 in one go)
* Support for Debian 12 will be enabled

**Comment by x-FBi** [comment]
Are you looking for any help?

**Comment by ddavness** [comment]
Currently working on merging v67 into the `current-dev` branch; thought there would be much more merge conflicts! :sweat_smile:

Biggest changes (and the largest source of merge conflicts) seem to be on the admin panel itself. Have to take a look on what changed from their side :)

**Comment by lifeboy** [comment]
I'm very happy to see your update, @ddavness! I wish I could be of more help though!  I've been reading the installation scripts to understand how this has all been put together and it's becoming clearer to me.  Hopefully I can contribute in the near future!  

I haven't given up on changing the database to pg either, so as soon as I understand things better, I'm going to give it a bash!

Thanks again for your effort!  Ko-fi bought.

**Comment by lifeboy** [comment]
Hi @ddavness. what is the status of the upgrade now?  It's been quite a while and getting an up to date version of NextCloud would really be great!
